packages/db/src/pouch/index.ts

```diff
@@ -0,0 +1,72 @@
+import path from "path";
+
+export * from "./types";
+
+import { DatabasesOptions } from "./databases";
+
+import { FSDatabases } from "./fs";
+import { MemoryDatabases } from "./memory";
+import { SqliteDatabases } from "./sqlite";
+
+import { Collections } from "@truffle/db/meta";
+import { Databases, Definitions } from "./types";
+
+export interface WorkspaceConfig {
+  workingDirectory?: string;
+  adapter?: {
+    name: string;
+    settings?: any; // to allow adapters to define any options type
+  };
+}
+
+export const forDefinitions = <C extends Collections>(
+  definitions: Definitions<C>
+) => (config: WorkspaceConfig): Databases<C> => {
+  const { WorkspaceDatabases, settings } = concretize<C>(config);
```

Comment:
Ditto `WorkspaceDatabases`

Reply:
Yeah, I don't like the name, but I couldn't think of anything better. Got any ideas?

Reply:
I just called it `constructor` :)